src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -2280,4 +2280,13 @@ private static String maybeQuoteETag(String eTag) {
     private static boolean startsWithIgnoreCase(String string, String prefix) {
         return string.toLowerCase().startsWith(prefix.toLowerCase());
     }
+
+    // Protects against illegal characters in blob names only if the client requests it
+    private String encodeBlob(String encodingType, String blobName) {
+        if (encodingType.equals("url")) {
```

Comment:
`Object.equals` checks its parameters against null but if `encodingType` is null then we throw a `NullPointerException` before calling `equals`.